### PR TITLE
[CommunicationAccessToken] Communication created version of access token 

### PR DESCRIPTION
**Pull request by JoshuaLai**
Currently, AzureCommunication depends on AzureCore's AccessToken. Chat is built in obj-c so requires an obj-c members. For chat to use AzureCommunication properly we need to create our own obj-c class access token. At the same time the Access token in Core can be changed back to a pure struct.

### CODE REVIEW

File: sdk/communication/AzureCommunication/Source/Authentication/CommunicationAccessToken.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import Foundation
+// This is class
```
Comment (sacheu) [inline sdk/communication/AzureCommunication/Source/Authentication/CommunicationAccessToken.swift:28]: remove this comment.

File: sdk/communication/AzureCommunication/Source/Authentication/CommunicationUserCredential.swift [hunk 2]
```diff
@@ -29,13 +29,13 @@ import AzureCore
...
 
-public typealias AccessTokenRefreshOnCompletion = (AccessToken?, Error?) -> Void
+public typealias AccessTokenRefreshOnCompletion = (CommunicationAccessToken?, Error?) -> Void
 public typealias TokenRefreshOnCompletion = (String?, Error?) -> Void
 
 /**
  The Azure Communication Services User token credential. This class is used to cache/refresh the access token required by Azure Communication Services.
 */
-@objcMembers public class CommunicationUserCredential: NSObject {
+public class CommunicationUserCredential: NSObject {
```
Comment (sacheu) [inline sdk/communication/AzureCommunication/Source/Authentication/CommunicationUserCredential.swift:38]: Why 'objcMembers' is removed?  i think this is needed for callingSDK to use this class.

Comment (JoshuaLai) [reply sdk/communication/AzureCommunication/Source/Authentication/CommunicationUserCredential.swift]: Yes sorry this was done by mistake

File: sdk/communication/AzureCommunication/Source/Authentication/CommunicationAccessToken.swift (hunk 1 above)
Comment (JoshuaLai) [reply sdk/communication/AzureCommunication/Source/Authentication/CommunicationAccessToken.swift]: Removed

**Review by tjprescott — COMMENTED** [review]
This looks good to me. Have you verified that the example app continues to compile with this change?

**Comment by RezaJooyandeh** [comment]
Is there any problem with keeping the annotation for AccessToken in Azure.Core? It is not scalable to duplicate Azure.Core types and I think it will lead us towards being less align with the Azure SDKs.

**Comment by bsiegel** [comment]
> Is there any problem with keeping the annotation for AccessToken in Azure.Core? It is not scalable to duplicate Azure.Core types and I think it will lead us towards being less align with the Azure SDKs.

We definitely understand the scalability issue, but unfortunately adapting Azure Core to be usable by ObjC libraries is not possible without significant re-engineering. We've been having discussions to determine the best way forward for this scenario, but we don't have a general solution yet, so we definitely want to revert this change for now.

**Comment by JoshuaLai** [comment]
> This looks good to me. Have you verified that the example app continues to compile with this change?
@tjprescott: I have been working with @raosanat and @sacheu to make sure their sample apps are still working

**Review by tjprescott — APPROVED** [review]

**Pull request merged**